protected/humhub/widgets/TimeAgo.php

```diff
@@ -42,56 +62,100 @@ public function init()
+    private function isRenderStatic()
     {
-        // Use static timeago
-        if (Yii::$app->params['formatter']['timeAgoStatic']) {
-            return '<span class="time"><span title="' . $this->getFullDateTime() . '">' . Yii::$app->formatter->asRelativeTime($this->timestamp) . '</span></span>';
-        }
+        $timeAgoStatic = $this->staticTimeAgo !== null ?  $this->staticTimeAgo : Yii::$app->params['formatter']['timeAgoStatic'];
+        return $timeAgoStatic;
+    }
 
-        // Convert timestamp to ISO 8601
-        $this->timestamp = date("c", $this->timestamp);
+    /**
+     * @return string
+     * @throws \yii\base\InvalidConfigException
+     */
+    private function renderStatic()
+    {
+        return '<span class="time"><span title="tt ' . $this->getFullDateTime() . '">' . Yii::$app->formatter->asRelativeTime($this->timestamp) . '</span></span>';
+    }
 
-        $this->getView()->registerJs('$(".time").timeago();', \yii\web\View::POS_END, 'timeago');
-        return '<span class="time" title="' . $this->timestamp . '">' . $this->getFullDateTime() . '</span>';
+    /**
+     * @param $elapsed
+     * @return bool
+     */
+    private function isTimeAgoElapsed($elapsed)
+    {
+        $timeAgoBefore = $this->timeAgoBefore !== null ? $this->timeAgoBefore : Yii::$app->params['formatter']['timeAgoBefore'];
+        return $timeAgoBefore !== false && $elapsed >= $timeAgoBefore;
+    }
+
+    /**
+     * @param $elapsed
+     * @return bool
+     */
+    private function isHideTimeAfter($elapsed)
+    {
+        $timeAgoHideTimeAfter = $this->hideTimeAfter !== null ? $this->hideTimeAfter : Yii::$app->params['formatter']['timeAgoHideTimeAfter'];
+        return $timeAgoHideTimeAfter === false || $elapsed >= $timeAgoHideTimeAfter;
     }
 
     /**
      * Show full date
-     * 
+     *
      * @param int $elasped time in seconds
      * @return string output of full date and time
+     * @throws \yii\base\InvalidConfigException
      */
     public function renderDateTime($elapsed)
     {
         // Show time when within specified range
-        if (Yii::$app->params['formatter']['timeAgoHideTimeAfter'] === false || $elapsed <= Yii::$app->params['formatter']['timeAgoHideTimeAfter']) {
+        if (!$this->isHideTimeAfter($elapsed)) {
             $date = $this->getFullDateTime();
         } else {
             $date = Yii::$app->formatter->asDate($this->timestamp, 'medium');
         }
 
-        return '<span class="time"><span title="' . $this->getFullDateTime() . '">' . $date . '</span></span>';
+        return '<span class="tt time"><span title="' . $this->getFullDateTime() . '">' . $date . '</span></span>';
+    }
+
+
+    /**
+     * Render TimeAgo Javascript
+     *
+     * @return string timeago span
+     * @throws \yii\base\InvalidConfigException
+     */
+    public function renderTimeAgo()
+    {
+       // $this->getView()->registerJs('$(".time").timeago();', \yii\web\View::POS_END, 'timeago');
```

Comment:
Can we remove this?

Reply:
Done, tests fail due to another problem which is fixed in https://github.com/humhub/humhub/pull/4478